### PR TITLE
Convert remote Thunderbird unit tests to pytest (#468)

### PR DESCRIPTION
**Pull request by venkateshpitta**

### CODE REVIEW

**Comment by venkateshpitta** [comment]
This PR is part of issue #486 to move Thunderbird unit tests to pytest

**Comment by venkateshpitta** [comment]
@whimboo please review the PR for issue #486

**Comment by venkateshpitta** [comment]
https://ci.appveyor.com/project/whimboo/mozdownload/build/1.0.278#L114 -- the failure happening is outside the submitted patch.. I can see it locally too.

File: tests/remote/test_thunderbird.py [hunk 1]
```diff
@@ -4,248 +4,124 @@
...
-                                     logger=self.logger,
-                                     **test['args'])
-
-    def test_tinderbox_scraper(self):
-        for test in tests_tinderbox_scraper:
-            mozdownload.TinderboxScraper(destination=self.temp_dir,
-                                         logger=self.logger,
-                                         **test['args'])
+@pytest.mark.parametrize("args,url", [
+    ({'application': 'thunderbird', 'platform': 'linux', 'version': '52.0'},
```
Comment (whimboo) [inline tests/remote/test_thunderbird.py]: If you could format the parameters similarly to the other ones below, that would be great.

File: tests/remote/test_thunderbird.py [hunk 2]
```diff
@@ -4,248 +4,124 @@
...
+        assert urllib.unquote(scraper.url) == urljoin(BASE_URL, url)
+
+@pytest.mark.parametrize("args,url", [
+    ({'application': 'thunderbird', 'platform': 'linux', 'version': '52.7.0'},
+     'thunderbird/candidates/52.7.0-candidates/build1/linux-i686/en-US/thunderbird-52.7.0.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'linux64', 'version': '52.7.0'},
+     'thunderbird/candidates/52.7.0-candidates/build1/linux-x86_64/en-US/thunderbird-52.7.0.tar.bz2'),
+    ({'application': 'thunderbird', 'platform': 'mac', 'version': '52.7.0'},
+     'thunderbird/candidates/52.7.0-candidates/build1/mac/en-US/Thunderbird 52.7.0.dmg'),
+    ({'application': 'thunderbird', 'platform': 'win32', 'version': '52.7.0', 'locale': 'en-GB'},
```
Comment (whimboo) [inline tests/remote/test_thunderbird.py]: Looks like we somehow lost the default locale check for win32.

File: tests/remote/test_thunderbird.py [hunk 3]
```diff
@@ -4,248 +4,124 @@
...
+    mozdownload.DailyScraper(destination=tmpdir, **args)
+
+@pytest.mark.parametrize("args", [
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central'}),
+    ({'application': 'thunderbird',
+      'platform': 'linux',
+      'branch': 'comm-central',
+      'extension': 'txt'}),
```
Comment (whimboo) [inline tests/remote/test_thunderbird.py]: That's a special test for the `.txt` extension. Please move this to the very end.

**Comment by whimboo** [comment]
Also don't worry about the failures. Those are expected, and we have to find a way to get around them.

**Comment by venkateshpitta** [comment]
@whimboo please review the edits based on review comments

**Comment by venkateshpitta** [comment]
@whimboo this is the complete patch for moving all of the uni tests from unittest to py.test.  Please review.

**Comment by whimboo** [comment]
@venkateshpitta I didn't expect that you upload all the changes as a single commit to this PR. I thought that we wanted to do it in separate PRs. Also this PR was close to be complete for landing.

Would you please be so kind and restore this PR to only cover the thunderbird remote test? Then please split the remaining changes per subfolder and create separate PRs as agreed on at the beginning? Reviewing such a massive patch would be hard. Thanks.

**Comment by whimboo** [comment]
As talked with @venkateshpitta on IRC I will wait with the review until the commit series has been cleaned-up. Here a suggestion to clean this up:

1) Create a new branch based on the `work_on_468` branch. This will make sure that all the commits also exist on the new branch
2) Do a diff against the `master` branch and export the content to a file. This will ensure that if something goes off, you have a backup of your work
3) Use `git rebase -i master` on the `work_on_468` branch, and remove all the commits not belonging to this PR
4) Push again to your remote repository to update this PR

**Comment by whimboo** [comment]
Hi @venkateshpitta. I haven't heard from you for a while and wonder if you had a bit of time to continue on this PR. Please let me know. Thanks!

**Comment by venkateshpitta** [comment]
Hi @whimboo I have removed the other commits by following the instructions.  Hoping everything is fine with the PR.  Please review and comment.

**Comment by whimboo** [comment]
Note that once you rearranged the lines we can land your patch! thanks a lot.

**Comment by whimboo** [comment]
@venkateshpitta would you mind finishing up this patch? It would be great to see it landed. Thanks!

**Pull request closed without merging**